lib/workspace.rb

```diff
@@ -0,0 +1,58 @@
+# workspace file for keeping all things slack
+require "dotenv"
+require "httparty"
+
+require_relative "user"
+require_relative "channels"
+
+Dotenv.load
+
+module Slack
+  class Workspace
+    URL = "https://slack.com/api/users.list"
+    attr_reader :users, :channels
+    attr_accessor :selected
+
+    def initialize
+      @users = User.list
+      @channels = Channel.list
+      @selected = nil
+    end
+
+    def select_user(user_descriptor)
+      users.find do |user|
+        if user_descriptor == user.name || user_descriptor == user.slack_id
+          @selected = user
+        end
+      end
+
+      if @selected == nil
+        return "Could not find a user with USER_NAME or SLACK_ID #{user_descriptor}."
+      end
+    end
+
+    def select_channel(channel_descriptor)
```

Comment:
See notes for `select_user`

Also if you notice the methods are identical... you could have one method that does both!